.github/workflows/generate_changelog.yml

```diff
@@ -4,17 +4,14 @@ on:
     types: [closed]
 jobs:
   GenerateChangelog:
-    if: github.event.pull_request.merged
+    if: github.event.pull_request.merged || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
       - run: cd tools && go install github.com/hashicorp/go-changelog/cmd/changelog-build
+      - run: ./scripts/generate-changelog.sh
       - run: |
-          ./scripts/generate-changelog.sh
-          git config --local user.email "changelogbot@hashicorp.com"
-          git config --local user.name "changelogbot"
-          git add CHANGELOG.md
-          git commit -m "Update CHANGELOG.md for #${{ github.event.pull_request.number }}" 
-          git push   
+          if [ ${{github.event_name == 'workflow_dispatch'}} ]; then MSG="Update CHANGELOG.md (Manual Trigger)"; else MSG="Update CHANGELOG.md for #${{ github.event.pull_request.number }}"; fi      
+          ./scripts/push-changes.sh "changelogbot@hashicorp.com" "changelogbot" "$MSG"
```

Comment:
Nit: Are we expecting anywhere else to run this script? Seems like more commonly a maintainer would manually doing the changelog update and committing with their Git configuration. (Nota bene: seems preferable to avoid shell script sprawl.)